frontend/test/pytest/test_from_plxpr.py

```diff
@@ -0,0 +1,453 @@
+
+        x, y, z = 0.9, 0.2, 0.5
+
+        qml.capture.enable()
+        plxpr = jax.make_jaxpr(circuit)(x, y, z)
+        qml.capture.disable()
+
+        converted = from_plxpr(plxpr)(x, y, z)
+
+        assert converted.eqns[0].primitive == catalyst.jax_primitives.func_p
+        assert converted.eqns[0].params["fn"] is circuit
+
+        catalyst_res = catalyst_execute_jaxpr(converted)(x, y, z)
+        assert len(catalyst_res) == 3
+
+        a = np.cos(y / 2) * np.exp(-0.5j * (x + z))
+        b = np.sin(y / 2) * np.exp(-0.5j * (x - z))
+        state = np.array([a, b])
+        expected_probs = np.abs(state) ** 2
+        expected_expval_x = np.conj(state) @ qml.X.compute_matrix() @ state
+        expected_expval_y = np.conj(state) @ qml.Y.compute_matrix() @ state
+        assert qml.math.allclose(catalyst_res[0], expected_expval_x)
+        assert qml.math.allclose(catalyst_res[1], expected_expval_y)
+        assert qml.math.allclose(catalyst_res[2], expected_probs)
+
+        qjit_obj = qml.qjit(circuit)
+        qjit_obj(x, y, z)
+        catalxpr = qjit_obj.jaxpr
+        call_jaxpr_pl = converted.eqns[0].params["call_jaxpr"]
+        call_jaxpr_c = catalxpr.eqns[0].params["call_jaxpr"]
+
+        compare_call_jaxprs(call_jaxpr_pl, call_jaxpr_c)
+
+
+class TestHybridPrograms:
+    """from_plxpr conversion tests for hybrid programs."""
+
+    def test_pre_post_processing(self):
+        """Test converting a workflow with pre and post processing."""
+
+        dev = qml.device("lightning.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit(x, y):
+            qml.RX(x, 0)
+            qml.RY(3 * y + 1, 1)
+            qml.CNOT((0, 1))
+            return qml.expval(qml.X(1)), qml.expval(qml.Y(0))
+
+        def workflow(z):
+            a, b = circuit(z, 2 * z)
+            return a + b
+
+        qml.capture.enable()
+        plxpr = jax.make_jaxpr(workflow)(0.5)
+        qml.capture.disable()
+
+        converted = from_plxpr(plxpr)(0.5)
+
+        res = catalyst_execute_jaxpr(converted)(0.5)
+
+        x = 0.5
+        y = 3 * 2 * 0.5 + 1
+
+        expval_x1 = np.sin(y)
+        expval_y0 = -np.sin(x) * np.sin(y)
+        expected = expval_x1 + expval_y0
+
+        assert qml.math.allclose(expected, res[0])
+
+        qjit_obj = qml.qjit(workflow)
+        qjit_obj(0.5)
+
+        call_jaxpr_pl = converted.eqns[1].params["call_jaxpr"]
+        call_jaxpr_c = qjit_obj.jaxpr.eqns[1].params["call_jaxpr"]
+
+        # qubit extraction and classical equations in a slightly different order
+        # thus cant check specific equations and have to discard comparing counts
+        compare_call_jaxprs(call_jaxpr_pl, call_jaxpr_c, skip_eqns=(4, 5, 6), skip_counts=True)
+        compare_eqns(call_jaxpr_pl.eqns[4], call_jaxpr_c.eqns[5], skip_counts=True)
+        compare_eqns(call_jaxpr_pl.eqns[5], call_jaxpr_c.eqns[6], skip_counts=True)
+        compare_eqns(call_jaxpr_pl.eqns[6], call_jaxpr_c.eqns[4], skip_counts=True)
```

Comment:
A test with two QNodes within one workflow might be nice to exclude cross-contamination of `_InterpreterState` etc.

Reply:
added